samcli/commands/check/resources/Pricing.py

```diff
@@ -0,0 +1,155 @@
+"""
+Pricing questions are asked here. Pricing is only done for
+Lambda Functions as of now. Data is stored in graph in
+Lambda function pricing object.
+"""
+
+from typing import List, Tuple
+import click
+
+from samcli.commands.check.resources.Graph import Graph
+from samcli.commands.check.resources.LambdaFunctionPricing import LambdaFunctionPricing
+from samcli.commands._utils.resources import AWS_LAMBDA_FUNCTION
+
+
+class Pricing:
+    _graph: Graph
+    _max_num_requests: int
+    _min_memory_amount: int
+    _max_memory_amount: int
+    _max_duration: int
+
+    def __init__(self, graph: Graph) -> None:
+        """
+        Args:
+            graph (Graph): The graph object. This is where all of the data is stored
+        """
+        self._graph: Graph = graph
+        self._max_num_requests: int = 1000000000000000000000
+        self._min_memory_amount: int = 128
+        self._max_memory_amount: int = 10000
+        self._max_duration: int = 900000
+
+    def ask_pricing_questions(self) -> None:
+        """
+        Pricing quetions for various resources get asked here
+        Pricing is only done for Lambda functions now
+        """
+        asked_lambda_questions = False
+        click.echo("Pricing Questions")
+        for resource in self._graph.resources_to_analyze:
+            # Only ask lambda quetions once for all lambda functions
+            if resource.resource_type == AWS_LAMBDA_FUNCTION and not asked_lambda_questions:
+                asked_lambda_questions = True
+                self._ask_lambda_function_questions()
+
+    def _ask_lambda_function_questions(self) -> None:
+        """Lambda function pricing questions"""
+
+        lambda_funciton_pricing = LambdaFunctionPricing()
+        user_input_requests = _ask(
+            "What are the total number of requests expected from all lambda functions in a given month?",
+            1,
+            self._max_num_requests,
+        )
+        lambda_funciton_pricing.number_of_requests = user_input_requests
```

Comment:
Actually, you kind of implemented what I asked in the previous comment. Since you have a class type called `LambdaFuctionPricing`, all Lambda related questions can be moved there, and you don't need to define it here. For instance, from here you can call;
```
lambda_funciton_pricing = LambdaFunctionPricing()
lambda_function_pricing.ask_questions()
```

and in the `LambdaFunctionPricing`, you can ask the actual questions;
```
def ask_questions():
        user_input_requests = _ask(
            "What are the total number of requests expected from all lambda functions in a given month?",
            1,
            self._max_num_requests,
        )
        self.number_of_requests = user_input_requests

        user_input_duration = _ask(
            "What is the expected average duration of all lambda functions (ms)?", 1, self._max_duration
        )
        self.average_duration = user_input_duration

        user_input_memory, user_input_unit = self._ask_memory(
            'Enter the amount of memory allocated (128MB - 10GB), followed by a ":", '
            "followed by a valid unit of measurement [MB|GB]"
        )
        self.allocated_memory = user_input_memory
        self.allocated_memory_unit = user_input_unit
```

Reply:
I agree. I will implement it that way.